### PR TITLE
Discord Bot: Remove everything referencing invites

### PR DESCRIPTION
**Pull request by rshipp**
### Description

Remove all discord-invite code.

### Motivation and Context

* Removes about 25% of the discord bot code we had to support.
* Massively reduces our Discord API usage to help prevent using up all our quota.

Task: https://app.asana.com/0/1201131148207877/1201625631650542/f

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/salesforce_api.rb [hunk 1]
```diff
@@ -476,7 +458,7 @@ def self.participant_to_struct(participant)
                       participant['ParticipantStatus'],
                       participant['CohortName'], participant['CohortScheduleDayTime'],
                       participant['CohortId'], participant['Id'],
-                      participant['DiscordInviteCode'], participant['DiscordUserId'], participant['DiscordServerId'],
+                      nil, participant['DiscordUserId'], participant['DiscordServerId'],
```
Comment (rshipp) [inline lib/salesforce_api.rb:461]: i left the attribute on the struct for now bc I don't know what might break if i remove it

File: lib/discord_bot.rb [hunk 2]
```diff
@@ -708,64 +539,6 @@ def list_misconfigured_command(event)
...
-      message_content << "\n❌ No Program found for this Discord server."
-      message_content << "\nFirst, sync this server with a Salesforce Program ID, like:"
-      message_content << "\n`#{BOT_COMMAND_KEY}#{ADMIN_COMMANDS[:sync_salesforce]} a2Y11000002HY5mEAX`"
-      message.edit(message_content)
-      return
-    end
-    Honeycomb.add_field('program.id', program['Id'])
-
-    message_content << "\n✅ Found Program #{program['Id']}..."
-    message.edit(message_content)
```
Comment (sadleb) [inline lib/discord_bot.rb:725]: Did you mean to remove this?

Comment (rshipp) [reply lib/discord_bot.rb:725]: yes, it was used to check participants for invite issues

Comment (sadleb) [reply lib/discord_bot.rb:725]: gotcha. I see now that it's also part of `sync_salesforce_command`. Just wanted to make sure we had this check where we needed it.

**Review by elenabaurkot — COMMENTED** [review]
lgtm!

**Pull request merged**